lib/semantics/check-do.cc

```diff
@@ -511,8 +500,10 @@ class DoContext {
     SymbolSet result;
     if (const auto *expr{GetExpr(expression)}) {
       for (const Symbol *symbol : evaluate::CollectSymbols(*expr)) {
-        if (const Symbol * root{GetAssociationRoot(*symbol)}) {
-          result.insert(root);
+        if (symbol) {
+          if (const Symbol * root{GetAssociationRoot(*symbol)}) {
```

Comment:
I believe that the set returned by `evaluate::CollectSymbols()` should never contain null pointers, so the check is not necessary. You could assert that's the case by doing `GetAssociateRoot(DEREF(symbol))`.

Reply:
Thanks, Tim.  Using ```DEREF()``` gives me a little peace of mind, since ```CollectSymbols()``` is implemented elsewhere.